core/include/scipp/core/element/rebin.h

```diff
@@ -42,6 +44,11 @@ static constexpr auto rebin = overloaded{
                             std::decay_t<decltype(data_old)>>) {
             data_new.value[inew] += data_old.value[iold] * scale;
             data_new.variance[inew] += data_old.variance[iold] * scale;
+          } else if constexpr (std::is_same_v<typename std::decay_t<decltype(
+                                                  data_new)>::value_type,
+                                              bool>) {
```

Comment:
@DavidFair I tried to make this shorter, but I think it may be wrong, should this be `std::span<bool>`?

Reply:
I think `bool` is correct, since we are asking the for the underlying type of the storage, and on MSVC it resolves the original warning where we were scaling a bool value

Reply:
Can you double check that please? I would expect this to be `std::span<bool>` -- the `value_type` should only take care of going from `ValueAndVariance<T>` to `T`, but `T` should be a span here?

Reply:
This resolves all but 2 warnings where `scale` is double and the element type is `float` causing a loss of precision warning

However, unlike the bool warning I am not worried about this, since the user is already using single precision float anyway so they wouldn't have seen the extra precision. Additionally, I believe this PR is now blocking @nvaytet 's work, so we can fix up the scale warning another time in the future

Reply:
So I went back and added :
`            static_assert(
                !std::is_same_v<std::decay_t<decltype(data_new)>::value_type,
                                span<bool>>);`

To the else branch, compilation still completed

Additionally, I added `            std::decay_t<decltype(data_new)>::value_type::foo = 1;` to the bool branch to force a compilation failure and got:

```
C:\Users\.\repos\scipp\core\include\scipp/core/element/rebin.h(53,1): fatal error C1001: An internal error has occurred in the compiler.
1>(compiler file 'msc1.cpp', line 1528)
1> To work around this problem, try simplifying or changing the program near the locations listed above.
1>Please choose the Technical Support command on the Visual C++
1> Help menu, or open the Technical Support help file for more information
1>C:\Users\.\repos\scipp\variable\include\scipp/variable/transform.h(208): message : see reference to function template instantiation 'auto scipp::core::element::<lambda_906e96fcc2f034b67e4ea7fc9536a848>::operator ()<T,scipp::span<const double,18446744073709551615>,scipp::span<const bool,18446744073709551615>,scipp::span<const double,18446744073709551615>>(const T &,const scipp::span<const double,18446744073709551615> &,const scipp::span<const bool,18446744073709551615> &,const scipp::span<const double,18446744073709551615> &) const' being compiled
1>        with
1>        [
1>            T=scipp::span<bool,18446744073709551615>
1>        ]
1>C:\Users\.\repos\scipp\variable\include\scipp/variable/transform.h(218): message : see reference to function template instantiation 'void scipp::variable::detail::call_in_place_impl<Op&,Indices,T&,scipp::core::ElementArrayView<const scipp::span<const double,18446744073709551615>>&,scipp::core::ElementArrayView<const scipp::span<const bool,18446744073709551615>>&,scipp::core::ElementArrayView<const scipp::span<const double,18446744073709551615>>&,0,1,2>(scipp::overloaded<scipp::core::element::<lambda_906e96fcc2f034b67e4ea7fc9536a848>,scipp::core::element::<lambda_975e399eed57490caf01ccb86de05e72>,scipp::core::transform_flags::<lambda_2435bc91d5b30410f9bfe00b2194ac97>,scipp::core::transform_flags::<lambda_f7d9b5e43e510067c1e711eeeacf4577>,scipp::core::transform_flags::<lambda_ba4a0b4ac796634d36814bae6d0ae6d8>>&,const Indices &,std::integer_sequence<unsigned __int64,0,1,2>,Arg,scipp::core::ElementArrayView<const scipp::span<const double,18446744073709551615>> &,scipp::core::ElementArrayView<const scipp::span<const bool,18446744073709551615>> &,scipp::core::ElementArrayView<const scipp::span<const double,18446744073709551615>> &)' being compiled
1>        with
1>        [
1>            Op=scipp::overloaded<scipp::core::element::<lambda_906e96fcc2f034b67e4ea7fc9536a848>,scipp::core::element::<lambda_975e399eed57490caf01ccb86de05e72>,scipp::core::transform_flags::<lambda_2435bc91d5b30410f9bfe00b2194ac97>,scipp::core::transform_flags::<lambda_f7d9b5e43e510067c1e711eeeacf4577>,scipp::core::transform_flags::<lambda_ba4a0b4ac796634d36814bae6d0ae6d8>>,
1>            Indices=std::tuple<scipp::core::ViewIndex,scipp::core::ViewIndex,scipp::core::ViewIndex,scipp::core::ViewIndex>,
1>            T=scipp::span<bool,18446744073709551615>,
1>            Arg=scipp::span<bool,18446744073709551615> &
1>        ]
```

So the compiler is correctly deducing that the underlying type is bool

Reply:
Oh, right, `bool` never has variances, so since the `if constexpr` above handles that case, we always get the actual element type here.